packages/@yoda/datacollection/index.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+var crypto = require('crypto')
+var https = require('https')
+var logger = require('logger')('datacollection')
+var DEFAULT_HOST = require('@yoda/env')().datacollection.uploadUrl
+// var DEFAULT_HOST = 'das-tc-service-test.rokid.com'
+var DEFAULT_URI = '/das-tracing-collection/tracingUpload'
+var property = require('@yoda/property')
+
+function _createMd5(rokidId, nonce) {
+  return crypto.createHash('md5')
+    .update(rokidId + 'apiVersion1.0.0' + 'nonce' + nonce + rokidId)
+    .digest('hex')
+    .toUpperCase()
+}
+
+/**
+  * Upload buried data
+  * @function upload
+  * @param {object Array} params - One or more detailed buried point data.
+  *  Must include the eventId, eventName, eventType.
+  *  - eventId: defined by the product.
+  *  - eventName: defined by the product.
+  *  - eventType: 0 basic data, 1 system events, 2 application events, 3 log event.
+  * @example
+  * dataCollection([{{"eventId":"datacollection-test","eventName":"datacollection-test","eventType":1,"extraKvs":{},"keys":{}}])
+  */
+function upload(params) {
+  logger.debug(JSON.stringify(params))
+  if (Object.prototype.toString.call(params) !== '[object Array]') {
+    logger.error('Error: params is not an object Array')
+    return
+  }
+  var rokidId = property.get('ro.boot.serialno')
+  var rokidDtId = property.get('ro.boot.devicetypeid')
```

Comment:
Naming is incompatible with others, please use deviceId and deviceTypeId, by the way, device type id need to fetch from cloud.